packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -1944,6 +1941,20 @@ operations will be generated by the CLI.`,
     ctx.mapResourceToStack(parent.name.value, resolverResourceId);
   }
 
+  private isSubscriptionOwnerArgumentOptional(rules: AuthRule[]) {
+    const ownerRules = rules.filter(rule => rule.allow === OWNER_AUTH_STRATEGY);
+
+    // if there are multiple owner rules then the owner argument is optional
+    if (ownerRules.length > 1) {
+      return true;
+    }
+
+    // if there are any public, private, and/or static group rules then the owner argument is optional
+    return rules.find(
```

Comment:
nit: maybe this function should return a boolean every time, so that it can return the value you want instead of having to do `this.isSubscriptionOwnerArgumentOptional(rules) ? false : true`?

Reply:
updated to Boolean, makes it cleaner, thanks!